usb: hid: Define sensor page usages IDs

zessadaoui opened the pull request:
This commit adds the following:
    * A subset of usage IDs defined in the Sensors' page of the HID standard.
    * A HID usage macro that works with IDs of 2 bytes.
    * A macro to define the exponent in a report descriptor.

The values and naming came from the following document: https://www.usb.org/sites/default/files/hutrr39b_0.pdf

Signed-off-by: Zak Essadaoui zak0@meta.com

kartben commented:
(misclicked, sorry, only meant to trigger your CI)

jfischer-no commented on include/zephyr/usb/class/hid.h:
```diff
@@ -141,6 +141,8 @@ extern "C" {
 /** HID Button Usage page */
 #define HID_USAGE_GEN_BUTTON		0x09
+/** HID Sensors Usage page */
+#define HID_USAGE_SENSORS               0x20
```
Please follow consistently style in the file and use tabs.

zessadaoui replied:
Originally I tried using tabs for consistency, but then I failed the local compliance checks, when I ran clang-format, it would replace tabs with spaces to make the compliance checks pass.
If I were to update this file, would you be able to override the compliance checks?

jfischer-no commented on include/zephyr/usb/class/hid.h:
```diff
@@ -357,12 +386,21 @@ extern "C" {
+ * @return    HID Usage Index item
+ */
+#define HID_USAGE16(idx)                                                                           \
+	HID_ITEM(HID_ITEM_TAG_USAGE, HID_ITEM_TYPE_LOCAL, 2), ((0xFF) & idx), ((0xFF) & (idx >> 8))
```
```c
#define HID_USAGE16(idx)			\
	HID_ITEM(HID_ITEM_TAG_USAGE, HID_ITEM_TYPE_LOCAL, 2), idx, (idx >> 8)
```

jfischer-no replied:
Fixed it for you.
```
#define HID_USAGE16(idx)			\
	HID_ITEM(HID_ITEM_TAG_USAGE, HID_ITEM_TYPE_LOCAL, 2), (uint8_t)idx, (idx >> 8)
```

zessadaoui committed:
usb: hid: Define sensor page usages IDs

This commit adds the following:
    * Sensor related usage IDs in the Sensors' page of the HID standard.
    * A HID usage macro that works with IDs of 2 bytes.
    * A macro to define the exponent in a report descriptor.

The values and naming came from the following document: https://www.usb.org/sites/default/files/hutrr39b_0.pdf

Signed-off-by: Zak Essadaoui <zak0@meta.com>

jfischer-no reviewed (approved):
Thanks
